### PR TITLE
chore: add more logging to k8s watcher stream

### PR DESCRIPTION
**Pull request by jakedipity**
Adds logging that captures the name, namespace, labels, and annotations of pods that are captured by the k8s watcher stream. This is to help debug why some people are experiencing an edge case where the agent stops picking up labels.

Ref: LOG-16194

### CODE REVIEW

File: common/k8s/src/middleware/metadata.rs [hunk 1]
```diff
@@ -100,13 +100,41 @@ impl K8sMetadata {
                 let obj_ref = ObjectRef::from_obj(&pod);
                 if reader.get(&obj_ref).is_none() {
                     Metrics::k8s().increment_creates();
+                    info!(
```
Comment (dkhokhlov) [inline common/k8s/src/middleware/metadata.rs:103]: info might be too chatty. agent sends these logs to backend too. affects the bill.

Comment (jakedipity) [reply common/k8s/src/middleware/metadata.rs]: I modified the code to track generic pod changes with the info level, and more detailed annotation/label information with the trace level.

I thought the metadata cache tracking might be good to track in general. Let me know if you think I should just make it all trace.

**Comment by jakedipity** [comment]
@dkhokhlov Updated the code to (hopefully) be more rustic? rusty?. Removed duplicate code and considered the organization chat we had about being more smart with types.

File: common/k8s/src/middleware/metadata.rs [hunk 2]
```diff
@@ -191,6 +215,25 @@ fn extract_image_name_and_tag(container_name: String, pod: &Pod) -> Option<MetaO
     None
 }
 
+fn log_watcher_pod(log_event: LogEvent, pod: &Pod) -> () {
+    info!(
```
Comment (dkhokhlov) [inline common/k8s/src/middleware/metadata.rs:219]: when you ran it - this "info" did not print many messages?

**Comment by jakedipity** [comment]
@dkhokhlov I was able to fix the linting and integration test issues in this PR which is on main now: https://github.com/logdna/logdna-agent-v2/pull/487

I'm going to remove your lint commit and rebase onto master.

Also I decided to change all logs to `trace`, I didn't see much noise on my cluster, but that's with only a single pod... probably better to err on the side of caution.

**Pull request merged**